App/Sagas/TextileEventsSagas.ts

```diff
@@ -46,53 +49,57 @@ function nodeEvents() {
   >(emitter => {
     const subscriptions: EventSubscription[] = []
     subscriptions.push(
-      Textile.events.addThreadUpdateReceivedListener(update => {
-        const { type_url } = update.payload
-        if (
-          type_url === '/Text' ||
-          type_url === '/Comment' ||
-          type_url === '/Like' ||
-          type_url === '/Files' ||
-          type_url === '/Ignore' ||
-          type_url === '/Join' ||
-          type_url === '/Leave'
-        ) {
-          emitter(groupActions.feed.refreshFeed.request({ id: update.thread }))
-          // FIXME: This is a hack. We need to examine the thread id and dispatch one or the other.
-          // Or this needs to send the whole Thread or at least the addition of key
-          emitter(photosActions.refreshPhotos.request(undefined))
-        }
+      Textile.events.addThreadUpdateReceivedListener(
+        (threadId, feedItemData) => {
+          if (
+            feedItemData.type === FeedItemType.Text ||
+            feedItemData.type === FeedItemType.Comment ||
+            feedItemData.type === FeedItemType.Like ||
+            feedItemData.type === FeedItemType.Files ||
+            feedItemData.type === FeedItemType.Ignore ||
+            feedItemData.type === FeedItemType.Join ||
+            feedItemData.type === FeedItemType.Leave
```

Comment:
super

Reply:
Yea, we're not using it in Photos, but theis `FeedItemData` type is especially powerful when you switch on it's `type` property... the compiler resolves the type of the associated `value` correctly.